### PR TITLE
Migrate from testdir to pytester

### PR DESCRIPTION
**Pull request by JosiasAurel**

### CODE REVIEW

File: testing/test_faulthandler.py [hunk 1]
```diff
@@ -108,21 +108,21 @@ def test_cancel_timeout_on_hook(monkeypatch, hook_name):
...
     """
     )
-    result = testdir.run(
+    result = pytester.run(
         sys.executable,
         "-X",
         "faulthandler",
         "-mpytest",
-        testdir.tmpdir,
+        pytester.tmpdir,
```
Comment (duthades) [inline testing/test_faulthandler.py:125]:
```suggestion
        pytester.path,
```

File: testing/test_faulthandler.py [hunk 2]
```diff
@@ -108,21 +108,21 @@ def test_cancel_timeout_on_hook(monkeypatch, hook_name):
 
 
 @pytest.mark.parametrize("faulthandler_timeout", [0, 2])
-def test_already_initialized(faulthandler_timeout, testdir):
+def test_already_initialized(faulthandler_timeout, pytester: Pytester):
```
Comment (duthades) [inline testing/test_faulthandler.py:111]:
```suggestion
def test_already_initialized(faulthandler_timeout: int, pytester: Pytester) -> None:
```

**Review by duthades — CHANGES_REQUESTED** [review]

**Comment by duthades** [comment]
Hi @JosiasAurel, thanks for contributing.

I have left some comments. It seems there are some test and linting errors. Nothing major. I suggest that you create an environment locally so that you run tests and linting checks and easily fix errors if any. You can find 'How to' details [here](https://github.com/pytest-dev/pytest/blob/master/CONTRIBUTING.rst#preparing-pull-requests) (It has both short and long version descriptions).

File: testing/test_faulthandler.py [hunk 3]
```diff
@@ -1,48 +1,48 @@
 import sys
 
 import pytest
+from _pytest.pytester import Pytester
```
Comment (duthades) [inline testing/test_faulthandler.py:4]: Add a line after this import.
```suggestion
from _pytest.pytester import Pytester

```

**Review by duthades — CHANGES_REQUESTED** [review]

**Review by bluetech — APPROVED** [review]
Thanks @JosiasAurel, this looks good! And thanks @duthades for the review.

**Pull request merged**